### PR TITLE
[Fix-10641] [Common] Fix the hdfs NPE

### LINKED ISSUE #10641: [Bug] [common] throw NullPointerException in HadoopUtils.java casue hdfs local mode is invalid.
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.

### What happened

image
this exception will cause the hdfs local mode is invalid.
image

### What you expected to happen

the hdfs local mode valid.

### How to reproduce

put up breakpoint on it and debug run the application.

### Anything else

I did not test whether non-local patterns would be affected by this bug. During the development process, the log output did not record the call stack of this NullPointerException which seriously affected my positioning problem! I don't know if it's the default logging level setting or something else, but I hope you can optimize your logging.

### Version

dev

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by 106umao**
Fixed #10641 

## Brief change log

Add null pointer special handling in HadoopUtils.java

## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

### CODE REVIEW

File: dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/utils/HadoopUtils.java [hunk 1]
```diff
@@ -129,6 +129,11 @@ private void init() throws NullPointerException {
             }
 
             String defaultFS = configuration.get(Constants.FS_DEFAULT_FS);
+
+            if (defaultFS == null) {
```
Comment (ruanwenjun) [inline dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/utils/HadoopUtils.java]: Can we optimize line 139 - 153? Some code is duplicate.

Comment (106umao) [reply dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/utils/HadoopUtils.java]: > Can we optimize line 139 - 153? Some code is duplicate.

👍 WIP.

Comment (106umao) [reply dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/utils/HadoopUtils.java]: Hi @ruanwenjun PTAL.

**Review by ruanwenjun — APPROVED** [review]
LGTM

**Pull request merged**